call_ins/inref/testcizhaltrcmain.c

```diff
@@ -0,0 +1,206 @@
+	status = ydb_ci("testcizhaltnoretv",1);
+	if (status)
+	{
+		printf("Return code: %d\n", status);
+		ydb_zstatus(errbuf, ERRBUF_SIZE);
+		printf("%s\n", errbuf);
+		fflush(stdout);
+	}
+	printf("main: Return from QUIT with (an unexpected integer) return value\n\n");
+
+	/* Now the same treatment for a string return */
+
+	printf("main: Initializing return value to 'snarf' for the HALT test\n");
+	retstrval.address = retbuf;
+	strcpy(retbuf, "snarf");
+	retstrval.length = sizeof("snarf") - 1;
+	status = ydb_ci("testcizhaltrcstr", &retstrval, useHALT);
+	if (status)
+	{
+		ydb_zstatus(errbuf, ERRBUF_SIZE);
+		printf("Return code: %d\n", status);
+		printf("%s\n", errbuf);
+		fflush(stdout);
+	}
+	printf("main: After HALT, return value is %.*s\n\n", (int)retstrval.length, retstrval.address);
+
+	printf("main: Initializing return value to 'snarf' for the ZHALT test\n");
+	retstrval.address = retbuf;
+	strcpy(retbuf, "snarf");
+	retstrval.length = sizeof("snarf") - 1;
+	status = ydb_ci("testcizhaltrcstr", &retstrval, useZHALT);
+	if (status)
+	{
+		ydb_zstatus(errbuf, ERRBUF_SIZE);
+		printf("Return code: %d\n", status);
+		printf("%s\n", errbuf);
+		fflush(stdout);
+	}
+	printf("main: After ZHALT, return value is %.*s\n\n", (int)retstrval.length, retstrval.address);
+
+	printf("main: Initializing return value to 'snarf' for the ZGOTO 0 test\n");
+	retstrval.address = retbuf;
+	strcpy(retbuf, "snarf");
+	retstrval.length = sizeof("snarf") - 1;
+	status = ydb_ci("testcizhaltrcstr", &retstrval, useZGOTO0);
+	if (status)
+	{
+		ydb_zstatus(errbuf, ERRBUF_SIZE);
+		printf("Return code: %d\n", status);
+		printf("%s\n", errbuf);
+		fflush(stdout);
+	}
+	printf("main: After ZGOTO 0, return value is %.*s\n\n", (int)retstrval.length, retstrval.address);
+
+	printf("main: Initializing return value to 'snarf' for the QUIT with no return value (but one expected) test\n");
+	retstrval.address = retbuf;
+	strcpy(retbuf, "snarf");
+	retstrval.length = sizeof("snarf") - 1;
+	status = ydb_ci("testcizhaltrcstr", &retstrval, useQUITretval);
+	if (status)
+	{
+		ydb_zstatus(errbuf, ERRBUF_SIZE);
+		printf("Return code: %d\n", status);
+		printf("%s\n", errbuf);
+		fflush(stdout);
+	}
+	printf("main: After QUIT with (an unexpected) return value, return value is %.*s\n\n",
+	       (int)retstrval.length, retstrval.address);
+
+	printf("main: Setting up for QUIT with (an unexpected string) return value test\n");
+	status = ydb_ci("testcizhaltnoretv",0);
+	if (status)
+	{
+		ydb_zstatus(errbuf, ERRBUF_SIZE);
+		printf("Return code: %d\n", status);
+		printf("%s\n", errbuf);
+		fflush(stdout);
+	}
+	printf("main: Return from QUIT with (an unexpected string) return value\n\n");
+
+	/* Now make a call with arguments to a routine that isn't expecting any - expect error */
+
+	printf("main: Initializing return value to -99 for the call with args to routine with no parms test\n");
+	retintval = -99;
+	status = ydb_ci("testcizhaltnoargs", &retintval, 42);
```

Comment:
testcizhaltnoargs is an M function that expects no arguments but call-in passes 1 argument and so we expect a FMLLSTMISSING error. Similarly, can we add a test case for when the M function expects 1 argument but the call-in passes 2 arguments. I would expect a similar error in that case too.

Reply:
Added .. we get ACTLSTTOOLONG in that case which seems proper. Now that I think of it, I believe there was an existing call_ins test that tested that already - yes, the argcnt subtest raises that error. But anyway, I've already added it so happy to leave it.